src/tags/jsonsort.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * @Author: RagingLink 
+ * @Date: 2020-07-28 21:09:40
+ * @Last Modified by: RagingLink
+ * @Last Modified time: 2020-08-01 20:2 7:36
+ *
+ * This project uses the AGPLv3 license. Please read the license file before using/adapting any of the code.
+ */
+
+const Builder = require('../structures/TagBuilder');
+
+module.exports =
+    Builder.ArrayTag('jsonsort')
+        .withAlias('jsort')
+        .withArgs(a => [a.require('array'), a.require('path'), a.optional('descending')])   
+        .withDesc('Sorts an array of objects based on on the provided `path`.\n' +
+            '`path` is a dot-noted series of properties.' +
+            'If `descending` is provided, sorts in descending order.\n' +
+            'If provided a variable, will modify the original `array`.'
+        ).withExample(
+            '{set;~array;{json;[\n  {"points" : 10, "name" : "Blargbot"},\n  {"points" : 3, "name" : "UNO"},\n' +
+            '  {"points" : 6, "name" : "Stupid cat"},\n  {"points" : 12, "name" : "Winner"}\n]}}\n' +
+            '{jsonstringify;{jsonsort;{slice;{get;~array};0};points};2}',
+            '[\n  "{\\"points\\":3,\\"name\\":\\"UNO\\"}",\n  "{\\"points\\":6,\\"name\\":\\"Stupid cat\\"}",' +
+            '\n  "{\\"points\\":10,\\"name\\":\\"Blargbot\\"}",\n  "{\\"points\\":12,\\"name\\":\\"Winner\\"}"\n]'
+        ).whenArgs('0-1', Builder.errors.notEnoughArguments)
+        .whenArgs('2-3', async (subtag, context, args) => {
+            let arr = await bu.getArray(context, args[0]),
+                path = args[1] ? args[1].split('.') : undefined;
+                descending = bu.parseBoolean(args[2]);
+            
+            if (!bu.isBoolean(descending))
+                descending = !!args[2];
+
+            if (arr == null || !Array.isArray(arr.v))
+                return Builder.errors.notAnArray(subtag, context);
+            if(!path) return Builder.errors.customError(subtag, context, 'No path provided');
+            //Map array with values of array[item][path]
+            let mappedArray = await Promise.all(arr.v.map(item => {
```

Comment:
your callback function does not return an array of Promises so I don't understand why you have a `Promise.all` here. Can you explain this choice? @RagingLink 

cc @Ratismal

Reply:
Yeah, this definitely doesn't need to be run with Promise.all